app/jobs/consolidated_item_email_job.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+class ConsolidatedItemEmailJob < ApplicationJob
+  # When the job runs, it will check emails that should have been sent within
+  # EMAIL_CHECK_INTERVAL.
+  # This also implies that the job should be configured to run at the same rate so items
+  # don't get missed and duplicate emails don't get sent.
+  EMAIL_CHECK_INTERVAL = 10.minutes
+
+  # Start with opening reminders.
+  def perform
+    generate_opening_email_items.each do |course, course_emails_hash|
+      course_emails_hash.each do |user_email, items_hash|
+        ConsolidatedMailer.opening_reminder_email(course, user_email, items_hash).deliver_later
+      end
+    end
+  end
+
+  private
+
+  # Generates a nested hash with the following structure for opening reminders:
+  # {
+  #   Course_1 => {
+  #     User::Email_1 => {
+  #       ActableType_1 => [ActableItems...],
+  #       ActableType_2 => [ActableItems...]
+  #     },
+  #     User::Email_2 => {
+  #       ActableType_1 => [ActableItems...],
+  #       ActableType_2 => [ActableItems...]
+  #     }
+  #   },
+  #   Course_2 => {...}
+  # }
+  #
+  # @return [Hash]
+  def generate_opening_email_items
+    ActsAsTenant.without_tenant do
+      opening_lesson_plan_items.each do |item|
+        next unless item.actable.include_in_consolidated_email?('opening')
+
+        item.course.users.each do |user|
+          emails_hash[item.course][user.primary_email_record][item.actable_type].push(item.actable)
+        end
+      end
+    end
+    emails_hash
+  end
+
+  # Items with start_at in the past will NOT have opening reminders.
+  # Students will need to rely on the Todo entries instead.
+  def opening_lesson_plan_items
+    datetime_range = (Time.zone.now - EMAIL_CHECK_INTERVAL)..Time.zone.now
+    Course::LessonPlan::Item.where(published: true, start_at: datetime_range).includes(:actable)
```

Comment:
I think we can shift the `where(published: true, start_at: datetime_range)` into a scope under `item`?